src/imitation/rewards/reward_net.py

```diff
@@ -171,170 +213,180 @@ def __init__(
-              observation inputs, with shape `(None,)`.
-            * layers is a dictionary mapping to individual checkpointable layers.
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        use_state: bool,
+        use_action: bool,
+        use_next_state: bool,
+        use_done: bool,
+        **kwargs,
+    ):
+        """Builds reward MLP.
+
+        Args:
+          observation_space: The observation space.
+          action_space: The action space.
+          use_state: should the current state be included as an input to the MLP?
+          use_action: should the current action be included as an input to the MLP?
+          use_next_state: should the next state be included as an input to the MLP?
+          use_done: should the "done" flag be included as an input to the MLP?
+          kwargs: passed straight through to build_mlp.
         """
-        pass
+        super().__init__()
+        combined_size = 0
 
-    def build_summaries(self):
-        super().build_summaries()
-        tf.summary.histogram("shaping_old", self._old_shaping_output)
-        tf.summary.histogram("shaping_new", self._new_shaping_output)
+        self.use_state = use_state
+        if self.use_state:
+            combined_size += get_flattened_obs_dim(observation_space)
 
+        self.use_action = use_action
+        if self.use_action:
+            combined_size += get_flattened_obs_dim(action_space)
 
-class BasicRewardNet(RewardNet, serialize.LayersSerializable):
-    """An unshaped reward net with simple, default settings.
+        self.use_next_state = use_next_state
+        if self.use_next_state:
+            combined_size += get_flattened_obs_dim(observation_space)
 
-    Intended to match the reward network trained for the original AIRL pendulum
-    experiments. Right now it has a linear function approximator for the theta network,
-    not sure if this is what I want.
-    """
+        self.use_done = use_done
+        if self.use_done:
+            combined_size += 1
+
+        full_build_mlp_kwargs = {
+            "in_size": combined_size,
+            "hid_sizes": (32, 32),
+            "out_size": 1,
+            "squeeze_output": True,
+        }
+        full_build_mlp_kwargs.update(kwargs)
+
+        self.mlp = networks.build_mlp(**full_build_mlp_kwargs)
+
+    def forward(self, state, action, next_state, done):
+        inputs = []
+        if self.use_state:
+            inputs.append(th.flatten(state, 1))
+        if self.use_action:
+            inputs.append(th.flatten(action, 1))
+        if self.use_next_state:
+            inputs.append(th.flatten(next_state, 1))
+        if self.use_done:
+            inputs.append(th.flatten(done, 1))
+
+        inputs_concat = th.cat(inputs, dim=1)
+
+        outputs = self.mlp(inputs_concat)
+        assert outputs.shape == state.shape[:1]
+
+        return outputs
+
+
+class BasicRewardNet(RewardNet):
```

Comment:
This class feels a bit superflous (basically just an adaptor for an MLP). Again perhaps we want to postpone this for another refactor though, but I'm inclined to at least a TODO here.

Reply:
Maybe we should make `RewardNetShaped` and `RewardNet` concrete, taking a `base_reward_net` and `potential_net` input? It doesn't feel like this loses much generality.

Reply:
Going to postpone this until I can clean up the entire reward function/discriminator setup.